### DISCUSSION TITLE
How does `ts_tree_edit` interact with `ts_parser_set_included_ranges`?

### DISCUSSION THREAD

**Discussion by jdtsmith** [tree-sitter/tree-sitter#2629] [Q&A]
If tree edits overlap within active included ranges, are those ranges expanded or contracted to incorporate the newly edited text?  Conversely, does updating the included range(s) to incorporate new edits re-parse the entire tree, or perform an incremental parse?  

Basically what's the best practice for:

1. Limiting tree-sitter to include only a sub-set of the text.
2. Being able to update and extend that text range as edits change them. 

For a concrete example, imagine limiting the range to the last several dozen lines of a file.  As that file is edited, `ts_tree_edit` sends edit updates.  How is/should the included_range be expanded to incorporate the new edits?

**Comment by sogaiu** [comment]
This is not an answer to the questions above, just a note to mention that a similar question came up at discord.

**Reply by jdtsmith** [reply]
Interesting. I think the answer is no, ranges are not expanded.  I've decided to use buffer narrowing instead.

**Reply by sogaiu** [reply]
Thanks for sharing!
